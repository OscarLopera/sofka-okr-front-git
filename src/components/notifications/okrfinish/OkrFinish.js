<<<<<<< HEAD
import React, { useState } from 'react';
import Modal from 'react-modal';
import '../../../assets/styles/notifications/notificationsWindows.scss';
import { auth } from '../../../functions/firebaseAuth';
import { useAuthState } from "react-firebase-hooks/auth";

const nameUser =  auth;

const OkrFinish = () => {
    const [modalIsOpen, setModalIsOpen] = useState(false)

    const [user] = useAuthState(auth);
    let nombreOkr = "Realizar Encuesta"

    const customStyles = {
        content: {
            top: '20%',
            left: '50%',
            right: 'auto',
            bottom: 'auto',
            marginRight: '-50%',
            width: '600px',
            transform: 'translate(-40%, -10%)',
            padding: '0',
        },
    };

    return (
        <div >
            <button onClick={() => setModalIsOpen(true)}>Terminar Okr</button>
            <Modal isOpen={modalIsOpen}
                onRequestClose={() => setModalIsOpen(false)}
                style={customStyles}>
                <header className='headerNotificationWindows' id='headerNotificationWindows'></header>
                <div className='bodyNotificationWindows' id='bodyNotificationWindows'>
                    <h1 className='h1NotificationWindows'>¡{user? nameUser.currentUser.displayName :""}! </h1>
                    <h2 className='h1NotificationWindows'>Terminaste el Okr {nombreOkr}</h2>
                    <p className='pNotificationWindows'>"Nuestra recompensa se encuentra en el esfuerzo y no en el
                    resultado, un esfuerzo total es una víctoria completa (Mahatma Gandhi)".</p>
                </div>
                <div className='imageNotificationWindows' id='imageNotificationWindows'>
                    <img src="https://www.sofka.com.co/wp-content/uploads/2020/08/sofka-logo-gradient-white.png" alt="logo-sofka"></img>
                </div>
                <footer className='footerNotificationWindows' id='footerNotificationWindows'></footer>
            </Modal>
        </div>
    )
}

=======
import React, { useState } from 'react';
import Modal from 'react-modal';
import '../notificationsWindows.scss'

const OkrFinish = () => {
    const [modalIsOpen, setModalIsOpen] = useState(false)
    let nombre = "Andres Caro"
    let nombreOkr = "Realizar Encuesta"

    const customStyles = {
        content: {
            top: '20%',
            left: '50%',
            right: 'auto',
            bottom: 'auto',
            marginRight: '-50%',
            width: '600px',
            transform: 'translate(-40%, -10%)',
            padding: '0',
        },
    };

    return (
        <div >
            <button onClick={() => setModalIsOpen(true)} id="buttonNotificationOkr">Terminar Okr</button>
            <Modal isOpen={modalIsOpen}
                onRequestClose={() => setModalIsOpen(false)}
                style={customStyles}>
                <header className='headerNotificationWindows' id='headerNotificationWindows'></header>
                <div className='bodyNotificationWindows' id='bodyNotificationWindows'>
                    <h1>¡ {nombre}! </h1>
                    <h2>Terminaste el Okr {nombreOkr}</h2>
                    <p>"Nuestra recompensa se encuentra en el esfuerzo y no en el
                    resultado, un esfuerzo total es una víctoria completa (Mahatma Gandhi)".</p>
                </div>
                <div className='imageNotificationWindows' id='imageNotificationWindows'>
                    <img src="https://www.sofka.com.co/wp-content/uploads/2020/08/sofka-logo-gradient-white.png" alt="logo-sofka"></img>
                </div>
                <footer className='footerNotificationWindows' id='footerNotificationWindows'></footer>
            </Modal>
        </div>
    )
}

>>>>>>> 25757de47d76f61e4cc1b301444fb005cb63181d
export default OkrFinish;