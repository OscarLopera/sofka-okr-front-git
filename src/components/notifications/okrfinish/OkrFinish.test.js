import React from 'react';
import Modal from 'react-modal';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import OkrFinish from './OkrFinish'
import sinon from 'sinon'

configure({ adapter: new Adapter() });

it('show content of modal', ()=>{
    const spy  = sinon.spy();
    const wrapper = shallow(<OkrFinish onTest={spy}></OkrFinish>)
    expect(spy.called)
    wrapper.unmount();
})

it('renders react-modal', () => {
    const wrapper = shallow(<OkrFinish />);
    expect(wrapper.find(Modal)).toHaveLength(1);
});

it('opens modal when button is clicked', () => {
    const wrapper = shallow(<OkrFinish />);
    expect(wrapper.find(Modal).prop('isOpen')).toBe(false);
    wrapper.find('button').simulate('click');
    expect(wrapper.find(Modal).prop('isOpen')).toBe(true);
});

it('renders childen when modal is open', () => {
    const wrapper = shallow(<OkrFinish>modal content</OkrFinish>);
    //expect(wrapper.find(Modal).prop('children')).toBe();
    expect(wrapper.find(Modal).prop('children')).toMatchSnapshot();
});