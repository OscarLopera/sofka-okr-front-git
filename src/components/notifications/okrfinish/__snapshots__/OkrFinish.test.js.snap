// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders childen when modal is open 1`] = `
Array [
  <header
    className="headerNotificationWindows"
    id="headerNotificationWindows"
  />,
  <div
    className="bodyNotificationWindows"
    id="bodyNotificationWindows"
  >
    <h1>
      ¡ 
      Andres Caro
      ! 
    </h1>
    <h2>
      Terminaste el Okr 
      Realizar Encuesta
    </h2>
    <p>
      "Nuestra recompensa se encuentra en el esfuerzo y no en el resultado, un esfuerzo total es una víctoria completa (Mahatma Gandhi)".
    </p>
  </div>,
  <div
    className="imageNotificationWindows"
    id="imageNotificationWindows"
  >
    <img
      alt="logo-sofka"
      src="https://www.sofka.com.co/wp-content/uploads/2020/08/sofka-logo-gradient-white.png"
    />
  </div>,
  <footer
    className="footerNotificationWindows"
    id="footerNotificationWindows"
  />,
]
`;
